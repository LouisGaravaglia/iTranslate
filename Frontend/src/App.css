html, body {
  min-width: 100vh;
  width: 100vw;
  height: 100vh;
  margin: 0px;
  padding: 0px !important;
  overflow-x: hidden; 
}

/* //////////////////////////////////////  NAVBAR  ////////////////////////////////////// */

.Navbar-Container-Left {
  position: absolute;
  height: 80px;
  /* border: solid 2px red; */
}

.Navbar-Container-Right {
  position: absolute;
  right: 0;
  height: 80px;
  /* border: solid 2px red; */
}

a {
  text-decoration: none;
  color: white;
  margin: 0 30px;
  font-weight: 500;
}

.Navbar-Search-Box {
  position: sticky;
  z-index: 4;
  top: 30px;
  margin: 0;
  padding: 0;
  margin-left: 50px;
  /* border: 2px solid magenta; */
}

.Navbar-Search {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 800;
  font-size: 1.5rem;
  color: #fff;
  margin: 0;
  padding: 0;
  /* margin-left: 50px; */
}

.Navbar-Browse-Box {
  position: sticky;
  z-index: 4;
  top: 30px;
  margin: 0;
  padding: 0;
  margin-right: 50px;
  /* border: 2px solid yellow; */
}

.Navbar-Browse {
  float: right;
  cursor: pointer;
  color: #fff;
}

/* //////////////////////////////////////  RETURN TO TOP ARROW  ////////////////////////////////////// */

.TopArrow-Container {
  position: relative;
  height: 150px;
  width: 100vw;
}

.TopArrow-Box {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  position: fixed;
  left:50%;
  transform: translate(-50%, -50%);
  bottom: 0;
  z-index: 4;
  text-decoration: none;
  width: 80px;
  height: 60px;
  cursor: pointer;
}


/* //////////////////////////////////////  BROWSE LANDING INDEX  ////////////////////////////////////// */

.Browse-Landing {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  width: 100vw;
  height: 100vh;
}

.Browse-Landing p {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 900;
  font-size: 4rem;
  margin: 0px 40px;
  padding: 6px 12px;
  flex: 0 1 100%;
}

/* //////////////////////////////////////  LYRICS  ////////////////////////////////////// */

.Lyrics-Translation {
  display: flex;
  flex-wrap: nowrap;
  justify-content: space-around;
  min-width: 100vw;
  min-height: 100vh;
  /* margin: 0 20px; */
}

.Lyrics-Container {
  /* background-color: #ef97bc; */
  margin: 0 20px 20px 50px;
  flex: 1 1 50vw;
  /* border: 2px solid #fff; */
}

.Lyrics {
  white-space: pre-line;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 600;
  font-size: 1.5rem;
  /* color: #522df1; */
  margin: 40px 40px;

}

/* //////////////////////////////////////  TRANSLATION  ////////////////////////////////////// */

.Translation-Container {
  /* background-color: #522df1; */
  margin: 0 50px 20px 20px;
  flex: 1 1 50vw;
  /* border: 2px solid #fff; */
}

.Translation {
  white-space: pre-line;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 600;
  font-size: 1.5rem;
  /* color: #fff; */
  margin: 40px 40px;
}

.inViewPlaceholder {
  height: 10px;
  width: 100vw;
}

/* //////////////////////////////////////  SEARCH BAR  ////////////////////////////////////// */

/* .Search-Home-Container {
    height: 100vh;
  width: 100vw;
background-color: rgb(2, 206, 169);
} */

.Search-Field {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  flex-direction: column;
  width: 100vw;
  height: 100vh;
}

.Search-Field-Content h1 {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 800;
  font-size: 5rem;
  display: block;
  text-align: center;
  cursor: default;
  /* margin: 0 80px; */
}

.Search-Field-Content {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  flex-direction: column;
  flex: 1;
  margin: 0 80px;
}

.Search-Field input {
  width: 40vw;
  height: 50px;
    border: none;
  /* border: 2px solid  #12006B;
  border-radius: 4px; */
}

.Search-Field button{
  background-color: #fff;
  border: none;
  font-size: 1rem;
  padding: 6px 12px;
  margin: auto;
  position: absolute;
  text-align: center;
  right:0;
  top:50%;
  transform: translate(-10%, -50%);
  /* height: 54px;
  width: 56px; */
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  cursor: pointer;
}



/* .Search-Field button:hover{
  background-color: rgb(58, 163, 255);
  cursor: pointer;
} */

.Search-Field-Filler {
flex: 0.1 0 1vh;
}

#SearchVal {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 400;
  font-size: 1.3rem;
  padding-left: 25px;
  outline:none;
}



.icon {
  font-size: 1.5rem;
}

.Search-Input-Container {
  position: relative;
}

/* //////////////////////////////////////  FLASH MESSAGE  ////////////////////////////////////// */

.Flash-Messages-Container {
  position: absolute;
  position: sticky;
  top: 0;
  z-index: 6;
}

.FlashMessage {
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: nowrap;
  background-color: #e767f0;
  height: 15vh;
  width: 100vw;
}

.FlashMessage p {
  color: white;
  margin: 0 30px;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 800;
  font-size: 2rem;

}

.FlashMessage span {
  color: white;
  margin: 0 30px;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 500;
  font-size: 2rem;
}

/* //////////////////////////////////////  DANCEABILITY ROUTE ////////////////////////////////////// */


/* DANCEABILITY */
.Browse-Danceability-v2 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  width: 100vw;
  height: 100vh;
  /* border: 2px solid turquoise; */
}

/* DANCEABILITY */
.Browse-Danceability-v2 h1{
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 900;
  font-size: 3rem;
  background: linear-gradient(to right, #4f1efe  0%, #5d3bc5  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  /* border: 2px solid blue; */
  text-align: center;
  margin: 20px 0 0;
}


/* DANCEABILITY - NO TRACKS */
.Pagination-Slider-Placeholder-v2 {
  /* border: 4px solid red; */
  width: 100vw;
  flex: 2;
}

/* DANCEABILITY - WITH TRACKS */
.Temp-v2 {
  /* border: 4px solid Purple; */
  width: 100vw;
  flex: 2;
}

.Danceability-Pagination-Slider-Container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  width: 100vw;
  flex: 2;
  /* border: 4px solid brown; */
}

.Danceability-Pagination-Slider {
  flex: 0 0 30vw;
}

/* DANCEABILITY */
.Danceability-Slider-Container-v2 {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  /* border: 2px solid yellow; */
  width: 100vw;
}

/* DANCEABILITY */
.Danceability-Slider-Item-v2 {
  flex: 0 0 80vw;
}

/* DANCEABILITY - NO TRACKS */
.Danceability-No-Results-Container {
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  width: 100vw;
  height: 100%;
  flex: 2;
  /* border: 2px solid black; */
}

.Danceability-No-Results {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 100;
  font-size: 5rem;
  background: linear-gradient(to right, #4f1efe  0%, #7c3bc5  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: default;
   /* border: 2px solid rgb(92, 255, 111); */
}

.Danceability-Result-Box {
  display: flex;
  justify-content: center;
  align-items: center;
  /* border: 2px solid green; */
  text-align: center;
  width: 100vw;
  flex: 2;
}

.Danceability-Result-Container {
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  /* border: 2px solid black; */
}

/* //////////////////////////////////////  RESULT (HEADER & SUBHEADER) ////////////////////////////////////// */

.Result-Box {
  border: 2px solid green;
  text-align: center;
  width: 100vw;
  flex: 4;
}

.Result-Header {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 100;
  font-size: 5rem;
  margin: 0;
  padding: 0;
}

.Result-Subheader {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 200;
  font-size: 2rem;
  cursor: default;
  margin: 0;
  padding: 0;
}

/* //////////////////////////////////////  MAIN CONTAINER ////////////////////////////////////// */

.Main-Container{
  position: relative;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  flex-wrap: wrap;
  width: 100vw;
  height: 100vh;
  border: 4px solid red;
}


.Browse-Albums {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  flex: 1 0 60vh;
  /* border: 4px solid yellow; */
}



.Main-Result-Container {
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  flex-direction: column;
  width: 100vw;
  height: 100%;
  /* border: 2px solid green; */
}

.Result-Main {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 100;
  font-size: 5rem;
  color: #fff;
  /* margin-bottom: 80px; */
  flex: 1 0 200px;
  /* border: 2px solid yellow; */
  margin: 0;
  padding: 0;
}

.Hover-Box {
  flex: 0 1 auto;
  /* border: 2px solid blue; */
  margin: 0;
  padding: 0;
}

.Result-Main-TopFiller {
flex: 10 1 auto;
/* border: 2px solid pink; */
}


.Main-Pagination-Slider-Container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  /* border: 4px solid #3e8e41; */
  width: 100vw;
  flex: 3 0 20vh;
}

.Main-Pagination-Slider {
  flex: 0 0 30vw;
}

.Main-Pagination-Slider-Placeholder {
  /* border: 4px solid #3e8e41; */
  text-align: center;
  flex: 2 0 20vh;
}

/* //////////////////////////////////////  ALBUM ////////////////////////////////////// */

.Album {
  position: relative;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  flex: 1 0 21%;
  margin: 40px;
}

.Album-Name-Box {
  position: relative;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  position: absolute;
  top:0;
  left: 0;
  height: 300px;
  width: 300px;
  background-color: #3800a7;
  opacity: 0.7;
  cursor: pointer;
}

.Album-Name {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 700;
  font-size: 2rem;
  /* color: rgb(255, 183, 95); */
  z-index: 6;
  margin: 5px 5px;
}

/* //////////////////////////////////////  LOADING ICON  ////////////////////////////////////// */

.Loading-Box {
  position: absolute;
  z-index: 2;
  left:50%;
  top:50%;
  transform: translate(-50%, -50%);
}

/* //////////////////////////////////////  SEARCH FONT COLORS  ////////////////////////////////////// */

.Search-Landing {
  background: linear-gradient(to right, #062bff 0%, #6000FF 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Search-Landing-Text {
  color: #A800FF;
}

.Search-Landing-Button {
  color: #A800FF;
}

.Search-Track-Subheader {
  background: linear-gradient(to right, #ff7ebc 0%, #ee67fa  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Search-Track-Header {
  background: linear-gradient(to right, #ff7ebc  0%, #ee67fa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Search-Language {
  background: linear-gradient(to right, #ee67fa 0%, #bf00ff 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Search-Language-Text {
  color: #6000FF;
}

.Search-Language-Button {
  color: #6000FF;
}

.Search-Lyrics {
  background: linear-gradient(to right, #ffc1ed 0%, #ee67fa  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Search-Translation {
  background: linear-gradient(to right, #fc92ec  0%, #ac37ff 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

/* //////////////////////////////////////  BROWSE BY GENRE FONT COLORS  ////////////////////////////////////// */

.Browse-Categories {
  background: linear-gradient(to right, #e767f0 0%,#aa35fe 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Genres {
  background: linear-gradient(to right, #ee6cd7 0%, #d11ff4 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Artists {
  background: linear-gradient(to right, #ffa114 0%, #d563c0 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Albums {
  background: linear-gradient(to right, #bf6571 0%, #ffb22e 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Tracks {
  background: linear-gradient(to right, #6a2fdc 0%, #8943b5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Language {
  background: linear-gradient(to right, #9161db 0%, #c594bf 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Genre-Language-Text {
  color: #e3b2af;
}

.Genre-Language-Button {
  color: #e3b2af;
}

.Genre-Lyrics {
  background: linear-gradient(to right, #4f1efe 0%, #7c3bc5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Translation {
  background: linear-gradient(to right, #7c3bc5 0%, #c1666f 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

/* //////////////////////////////////////  BROWSE BY ARTISTS FONT COLORS  ////////////////////////////////////// */

.Artist-Artists {
  background: linear-gradient(to right, #fdc3df 0%, #b84dfa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Artist-Albums {
  background: linear-gradient(to right, #fdc3df 0%, #b84dfa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Artist-Tracks {
  background: linear-gradient(to right, #6a2fdc 0%, #8943b5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Artist-Language {
  background: linear-gradient(to right, #bf00ff 0%, #e167fa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Artist-Language-Text {
  color: #dba3e6;
}

.Artist-Language-Button {
  color: #dba3e6;
}

.Artist-Lyrics {
  background: linear-gradient(to right, #4f1efe 0%, #7c3bc5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Artist-Translation {
  background: linear-gradient(to right, #7c3bc5 0%, #ce5b93 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

/* //////////////////////////////////////  BROWSE BY DANCEABILITY FONT COLORS  ////////////////////////////////////// */

.Danceability-Track-Subheader {
  background: linear-gradient(to right, #4f1efe  0%, #7c3bc5  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Danceability-Track-Header {
  background: linear-gradient(to right, #4f1efe 0%, #7c3bc5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Danceability-Language {
  background: linear-gradient(to right, #7c3bc5 0%, #d383fa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Danceability-Language-Text {
  color: #9eb5fa;
}

.Danceability-Language-Button {
  color: #9eb5fa;
}

.Danceability-Lyrics {
  background: linear-gradient(to right, #4f1efe 0%, #7c3bc5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Danceability-Translation {
  background: linear-gradient(to right, #7c3bc5 0%, #f263a8 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}