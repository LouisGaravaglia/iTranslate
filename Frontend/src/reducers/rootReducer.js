import {combineReducers} from "redux";
import translationReducer from "./translationReducer";
import errorsReducer from "./errorsReducer";
import lyricsReducer from "./lyricsReducer";
import resultsReducer from "./resultsReducer";

const rootReducer = combineReducers({results: resultsReducer, translation: translationReducer, lyrics: lyricsReducer, errors: errorsReducer});

export default rootReducer;