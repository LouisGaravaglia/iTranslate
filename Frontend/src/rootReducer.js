// const SEED_ARTISTS = [{name: "J Balvin", id:"1vyhD5VmyZ7KMfW5gqLgo5"}, {name: "Bad Bunny", id:"4q3ewBCX7sLwd24euuV69X"}, {name: "Maluma", id:"1r4hJ1h58CWwUQe3MxPuau"}, {name: "Rosal√≠a", id:"7ltDVBr6mKbRvohxheJ9h1"}];
// const INITIAL_STATE = {songs:[], artists: SEED_ARTISTS, albums:[]};
import {GET_TRANSLATION} from "./actionTypes";
const ERRORS = {searchError: false, lyricsError: false, translationError: false}
const INITIAL_STATE = {translation: "", errors: ERRORS }

function rootReducer(state=INITIAL_STATE, action) {
    switch(action.type) {
        case GET_TRANSLATION:
        
            // const updatedStateWithErrors = {...state, state.errors.lyricsError = action.errors.lyricsError}
        return {...state, translation: action.translation}

        default:
            return state;
    }
}

export default rootReducer;
