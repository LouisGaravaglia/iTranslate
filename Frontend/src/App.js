import React, {useEffect, useState} from 'react';
import './App.css';
import Routes from "./Routes";
import Navbar from "./Navbar";
import UserContext from "./UserContext";
import IBMWatsonAPI from "./IBMWatsonAPI";


function App() {
  const [languages, setLanguages] = useState("");

  //GET AVAILABLE LANGUAGES TO TRANSLATE LYRICS TO FROM IBM API
  useEffect(() => {
    async function getLanguages() {
      const res = await IBMWatsonAPI.getLanguages();
      console.log("my languages!!!: ", res);
      setLanguages(res);
    }
    getLanguages();
  }, []);


  return (
  <UserContext.Provider value={{languages}}>
    <div className="App">
      <Navbar />
      <Routes />
    </div>
  </UserContext.Provider>
  );
}

export default App;
