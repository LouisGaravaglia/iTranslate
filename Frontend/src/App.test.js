import React from 'react'
import {render, fireEvent, waitFor, screen, cleanup, getByTestId} from '@testing-library/react'
import {Router, MemoryRouter} from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux'
import { createStore } from "redux";
import { createMemoryHistory } from 'history'
import rootReducer from "./reducers/rootReducer";
import App from "./App";

afterEach(cleanup);

const mockIntersectionObserver = class {
  constructor() {}
  observe() {}
  unobserve() {}
  disconnect() {}
};

beforeEach(() => {
  window.IntersectionObserver = mockIntersectionObserver;
});

const startingState = {
  selectedTrack: {trackName: "chanel", trackId: 42}, 
  errors: {translationError: false, languageError: false, lyricsError: false, searchError: false, generalError: false}
};

//REDUCER FUNCTION TO MIMIC REDUX REDUCER FOR STORE
function reducer(state = startingState, action) {
  switch (action.type) {
    case "ADD_ALBUMs":
      return {...state, albums: action.albums};
    default:
      return state;
  };
};

//FUNCTION TO ALLOW THE COMPONENT TO BE RENDERED USING OUR MAKESHIF REDUX STORE
function renderWithRedux(
  component,
  {initialState, store = createStore(reducer, initialState)} = {}
) {
  return {
    ...render(<Provider store={store}>{component}</Provider>)
  };
};

//SMOKE TEST
describe('Smoke Test for App component', () => {

  it('renders without crashing', () => {
    const history = createMemoryHistory();

    const {getByText} = renderWithRedux(
      <Router history={history}>
        <App />
      </Router>
    );
  });
});

//SNAPSHOT TEST
describe('Snapshot Test for App component', () => {

  it('matches snapshot', () => {
    const history = createMemoryHistory();

    const {asFragment} = renderWithRedux(
      <Router history={history}>
        <App />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

});


describe('Make sure appropriate text is appearing in certain routes', () => {

  it('displays the "find your song!" search bar', () => {

    const {getByText} = renderWithRedux(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );

    expect(getByText("Find your song!")).toBeInTheDocument();
  });

  it('displays the "Artists Genre Danceability" in the Browse route', () => {

    const {getByText} = renderWithRedux(
      <MemoryRouter initialEntries={["/browse"]}>
        <App />
      </MemoryRouter>
    );

    expect(getByText("Artists")).toBeInTheDocument();
  });

});