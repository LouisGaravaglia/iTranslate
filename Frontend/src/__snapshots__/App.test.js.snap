// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Navbar-Container"
    >
      <div
        class="Navbar"
      >
        <div
          class="Navbar-Search-Box"
        >
          <a
            aria-current="page"
            class="Navbar-Search active"
            href="/"
          >
            LYRCS
          </a>
        </div>
        <div
          class="Navbar-Browse-Box"
        >
          <svg
            class="sc-bdVaJa fUuvxv Navbar-Browse"
            fill="#fff"
            height="40px"
            rotate="0"
            viewBox="0 0 1024 1024"
            width="40px"
          >
            <path
              d="M569.8 825.2l276.2-268c12-11.6 18-27.4 18-44.8v-0.8c0-17.4-6-33.2-18-44.8l-276.2-268c-24-25-62.6-25-86.4 0s-23.8 65.4 0 90.4l166 158.8h-428c-34-0-61.4 28.6-61.4 64 0 36 27.4 64 61.2 64h428l-166 158.8c-23.8 25-23.8 65.4 0 90.4 24 25 62.6 25 86.6 0z"
            />
          </svg>
        </div>
        <a
          href="/browse"
        />
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="Flash-Messages-Container"
      />
      <div
        style="background-color: rgb(29, 77, 190);"
      >
        <div
          style="opacity: 1; margin-left: -2500px;"
        >
          <div
            class="Search-Field"
          >
            <div
              class="Search-Field-Content"
            >
              <h1>
                Find your song!
              </h1>
              <form
                class="Search-Input-Container"
              >
                <div>
                  <input
                    id="SearchVal"
                    name="searchVal"
                    type="text"
                    value=""
                  />
                  <button
                    type="submit"
                  >
                    <i
                      class="fa fa-search icon"
                    />
                  </button>
                </div>
              </form>
            </div>
            <div
              class="Search-Field-Filler"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
